enterprise/models.py

```diff
@@ -4198,3 +4198,61 @@ def submit_for_configuration(self, updating_existing_record=False):
         self.submitted_at = localized_utcnow()
         self.save()
         return sp_metadata_url
+
+
+class EnterpriseGroup(TimeStampedModel):
+    """
+    Enterprise Group model
+
+    .. no_pii:
+    """
+    uuid = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    name = models.CharField(
+        max_length=25,
+        blank=False,
+        help_text=_(
+            'Specifies enterprise group name.'
+        )
+    )
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        blank=False,
+        null=False,
+        related_name='groups',
+        on_delete=models.deletion.CASCADE
+    )
+    history = HistoricalRecords()
+    unique_together = (("name", "enterprise_customer"),)
+
+
+class EnterpriseGroupMembership(TimeStampedModel):
+    """
+    Enterprise Group Membership model
+
+    .. no_pii:
+    """
+    uuid = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    group = models.ForeignKey(
+        EnterpriseGroup,
+        blank=False,
+        null=False,
+        related_name='members',
+        on_delete=models.deletion.CASCADE,
+    )
+    enterprise_customer_user = models.ForeignKey(
+        EnterpriseCustomerUser,
+        blank=True,
+        null=True,
+        related_name='memberships',
+        on_delete=models.deletion.CASCADE,
+    )
+    pending_enterprise_customer_user = models.ForeignKey(
+        PendingEnterpriseCustomerUser,
+        blank=True,
+        null=True,
+        related_name='memberships',
+        on_delete=models.deletion.CASCADE,
+    )
+    history = HistoricalRecords()
+    # https://code.djangoproject.com/ticket/9039
```

Comment:
Why are we referencing this django issue?

Reply:
I was worried that a unique constraint on fields that are null would be an issue - ie "this customer & this field = null is not unique therefore blow up" But I believe that issue is addressed by the django issue. I can be more clear as to the reason behind the comment!